### DISCUSSION TITLE
Status LEDs toggle on shutdown/boot

### DISCUSSION THREAD

**Discussion by Darkbluetest** [seerge/g-helper#3354] [Ideas]
Hello

Just what title says, could it be possible to add an option that turns off the status leds when system is shutting down, and turn them on, on boot? This would be useful for people sleeping in the same room (lol), for example, but that wants to see the battery status, disk access etc in the status leds when using it.

Tx!

**Comment by seerge** [comment]
@Darkbluetest hello, unfortunately that's not possible unless you have a Strix model from 2024. There Asus has added a separate toggle for status LEDs, and you can already access it via `Extra` settings in the app.

**Comment by Darkbluetest** [comment]
Yes, I have a Strix G18 from 2024. And the option in Extra settings works for me, I can turn it on or off. 

I was wondering only to make it automatic (or add an option): off when shutting down, on when login. It seems to me that if the model supports it, which in a Strix from 2024 is supported, then to add this automatism should be possible.

**Reply by seerge** [reply]
@Darkbluetest it may be quite tricky to do, what is the practical purpose of that ? Why don't you completely turn them OFF if you don't need them ?

**Comment by Darkbluetest** [comment]
I prefer to have the lights on while I use the PC, if I have them turned off always, I don't know when it is accessing disk, and if it is coherent with what I do/see in the OS, neither if the battery is charging/charged or not, regardless of what the OS says. It is a double check, and a feedback by a secondary/alternate way. Is like having a compass in the car, some people don't see the need, for me is invaluable regardless of how many GPS you may use.

In my case, for example, I have the laptop in a living room (practically where I do everything, almost). At night I have to close the door, because (at least in 2024 strix) the power led is on -I don't want to unplug it every night, the wall plug is not easily reachable, and the laptop DC plug, I don't want to make it suffer each day, I had problems with other laptop for that- and incredible as it may sound it lights all the living room! Yes, at least in the Strix 2024 is so bright and big. Imagine for people sleeping in the same room...

Of course I could live with it as it is, but I think it would be nice if possible. That depends how difficult/reliable it could be...

**Comment by Valour549** [comment]
+1 to this idea.

Status LED indicators are nice to use when the laptop is on. But as long as the laptop is plugged in, the battery indicator will always shine when the laptop is off. But indeed the implementation would be a lil tricky having to detect shutdown/startup (or log off/log on) to automate the process.

@Darkbluetest What I have done atm is bind them to my M4 and M5 hotkey like this. You can get the LED off and LED on exes from [here](https://github.com/Valour549/Scar18-LED-Toggle) and bind them so it becomes super easy to toggle. Note that at some point Windows Defender will probably tell you it's a virus (I compiled it myself on my PC and my Windows Defender tells me it's a trojan lol), so just add them to exceptions.

<img width="1280" alt="image" src="https://github.com/user-attachments/assets/ac15515f-e165-44b5-ae19-f812c2a935b7">

**Comment by seerge** [comment]
@Darkbluetest @Valour549 try this build. There is no UI toggle here (yet), as I'm not sure how popular this setting could be, but I have added an optional cofig param.

Quit app, add following to `%appdata%\GHelper\config.json` 

```
"auto_status_led": 1,
```

And start app again.
[GHelper.zip](https://github.com/user-attachments/files/17704148/GHelper.zip)

It will turn LED OFF on Shutdown and Hibernate (if that happens when laptop is awake and G-Helper is running and not suspended), and set LED ON when G-Helper starts or detects wake up from sleep

**Comment by Valour549** [comment]
@seerge Testing on my Scar 18 produces expected results upon startup/shutdown. Activating hibernation and sleep also works to turn off LEDs, but waking up from hibernation or sleep does not turn on LEDs. [Video here](https://www.youtube.com/watch?v=XEOyILMpRJs).

Note that I have [disabled Modern Standby](https://www.elevenforum.com/t/disable-modern-standby-in-windows-10-and-windows-11.3929/) and I am using the traditional sleep and hibernation.

![Screenshot 2024-11-11 230157](https://github.com/user-attachments/assets/6e7bf0f2-9c16-4931-9719-35cae39d04a6)

**Reply by seerge** [reply]
@Valour549 it will be more handy if you would just upload app log :)

**Comment by seerge** [comment]
@Valour549 and check if this build makes a difference
[GHelper.zip](https://github.com/user-attachments/files/17704140/GHelper.zip)

**Reply by Valour549** [reply]
Auto LED indicator is working all correctly now. But while testing, I inadvertently found a bug with miniLED mode not being auto applied when waking up from Hibernation (it applies correctly upon reboot and waking up from sleep).

[log.txt](https://github.com/user-attachments/files/17704715/log.txt)

**Reply by seerge** [reply]
@Valour549 hmm, ok, this build should send your mini-led init on every wake up as well.

[GHelper.zip](https://github.com/user-attachments/files/17704908/GHelper.zip)

**Reply by Valour549** [reply]
Fixed, great!

**Reply by seerge** [reply]
Just in case it's a firmware issue in a first place, app offers a workaround ...

**Reply by Valour549** [reply]
Yeah, the bug where the incorrect miniLED mode is applied upon reboot or waking from hibernation is a vanilla bug. There're actually quite a few other bugs regarding HDR on Asus laptops, but that's well outside the scope of G-Helper.

**Comment by Darkbluetest** [comment]
Hi. Thanks a lot for this. It works perfectly in my case. It does just what I thought. Thanks too to Valour549 for all testing and feedback.
Tx!

**Reply by seerge** [reply]
Great, it’s already live in 0.199
